src/main/kotlin/com/sourcegraph/cody/chat/ChatSession.kt

```diff
@@ -0,0 +1,224 @@
+package com.sourcegraph.cody.chat
+
+import CodyAgent
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.xml.util.XmlStringUtil
+import com.jetbrains.rd.util.AtomicReference
+import com.sourcegraph.cody.CodyToolWindowContent
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.ExtensionMessage
+import com.sourcegraph.cody.agent.WebviewMessage
+import com.sourcegraph.cody.agent.protocol.ChatMessage
+import com.sourcegraph.cody.agent.protocol.ChatRestoreParams
+import com.sourcegraph.cody.agent.protocol.ChatSubmitMessageParams
+import com.sourcegraph.cody.agent.protocol.Speaker
+import com.sourcegraph.cody.chat.ui.ChatPanel
+import com.sourcegraph.cody.commands.CommandId
+import com.sourcegraph.cody.config.RateLimitStateManager
+import com.sourcegraph.cody.vscode.CancellationToken
+import com.sourcegraph.common.CodyBundle
+import com.sourcegraph.common.CodyBundle.fmt
+import com.sourcegraph.common.UpgradeToCodyProNotification.Companion.isCodyProJetbrains
+import com.sourcegraph.telemetry.GraphQlLogger
+import io.ktor.util.collections.*
+import org.slf4j.LoggerFactory
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ExecutionException
+import javax.annotation.concurrent.GuardedBy
+
+typealias SessionId = String
+
+interface ChatSession {
+
+  fun sendMessage(text: String)
+
+  fun receiveMessage(extensionMessage: ExtensionMessage)
+
+  fun getCancellationToken(): CancellationToken
+}
+
+interface ChatSessionComponent {
+  fun registerCancellationToken(cancellationToken: CancellationToken)
```

Comment:
I'm not sure what's the purpose of this interface. This abstraction is not used anywhere, apart from forcing us to 'registerCancellationToken' to have the same name everywhere (which, imho, is not a sufficient argument for introducing such abstraction).

(side note: ChatPanel also has registerCancellationToken method, but it does not implement this interface)

Reply:
Agree, removed.